pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -934,31 +986,39 @@ func (cc *XController) manageQueueJob(qj *arbv1.AppWrapper) error {
 					}
 				}
 			}
-			if glog.V(10) {
-				if (is_first_deployment) {
-					current_time := time.Now()
-					glog.V(10).Infof("[TTime] %s, %s: WorkerBeforeCreatingResouces", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
-					glog.V(10).Infof("[Agent Controller] XQJ %s has Overhead Before Creating Resouces: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
-				}
-			}
+//			if glog.V(10) {
+//				if (is_first_deployment) {
+//					current_time := time.Now()
+//					glog.V(10).Infof("[TTime] %s, %s: WorkerBeforeCreatingResouces", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
+//					glog.V(10).Infof("[Agent Controller] XQJ %s has Overhead Before Creating Resouces: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
+//				}
+//			}
+			glog.V(4).Infof("[worker-manageQJ] %s 3Delay=%s BeforeDispatchingToEtcd version=%s Status=%+v",
+				qj.Name, metav1.Now().Sub(qj.Status.ControllerFirstTimestamp.Time), qj.ResourceVersion, qj.Status)
+			success := false
 			for _, ar := range qj.Spec.AggrResources.Items {
 				err00 := cc.qjobResControls[ar.Type].SyncQueueJob(qj, &ar)
 				if err00 != nil {
 					glog.V(4).Infof("Error from sync job: %v", err00)
+				} else {
+					success = true
 				}
 			}
-			if glog.V(10) {
-				if (is_first_deployment) {
-					current_time := time.Now()
-					glog.V(10).Infof("[TTime] %s, %s: WorkerAfterCreatingResouces", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
-					glog.V(10).Infof("[Agent Controller] XQJ %s has Overhead After Creating Resouces: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
-				}
+			if success { // set QueueJobStateRunning if at least one resource is successfully dispatched
+				qj.Status.QueueJobState = arbv1.QueueJobStateDispatched
+				glog.V(4).Infof("[worker-manageQJ] %s 4Delay=%s AtLeastOneResourceDispatchedToEtcd version=%s Status=%+v",
+					qj.Name, metav1.Now().Sub(qj.Status.ControllerFirstTimestamp.Time), qj.ResourceVersion, qj.Status)
 			}
+//			if glog.V(10) {
+//				if (is_first_deployment) {
+//					current_time := time.Now()
+//					glog.V(10).Infof("[Agent Controller] XQJ %s has Overhead After Creating Resouces: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
+//				}
+//			}
 
 			// TODO(k82cn): replaced it with `UpdateStatus`
-			if _, err := cc.arbclients.ArbV1().AppWrappers(qj.Namespace).Update(qj); err != nil {
-				glog.Errorf("Failed to update status of AppWrapper %v/%v: %v",
-					qj.Namespace, qj.Name, err)
+			qj.Status.FilterIgnore = true  // update State & QueueJobState after dispatch
```

Comment:
@zhangliresearch: Could you add a comment above indicating this is a todo so we can identify this is a workaround. E.g. //todo: This is a current workaround for duplicate message bug. Thank you :)

Reply:
This is not to address the duplicate message bug.

Setting FilterIgnore=true means this update will not generate an event for the eventQueue.  After dispatching qj to etcd for Kubernetes to work on, the only thing left is to find out if the pods are running.  This will be handled by updates of the ClusterStateCache every second. Getting the pod running could take some time.  There is no need to generate an event to the eventQueue right away.

Reply:
Got it.  Thanks for the clarification.